src/pubsub.c

```diff
@@ -353,18 +353,34 @@ int pubsubPublishMessage(robj *channel, robj *message) {
     return receivers;
 }
 
+/* This wraps handling ACL channel permissions for the given client. */
+int pubsubCheckACLPermissions(client *c, int idx, int count, int literal) {
+    /* Check if the user can run the command according to the current
+     * ACLs. */
+    int acl_chanpos;
+    int acl_retval = ACLCheckPubsubPerm(c,idx,count,literal,&acl_chanpos);
+    if (acl_retval == ACL_DENIED_CHANNEL) {
+        addACLLogEntry(c,acl_retval,acl_chanpos,NULL);
+        rejectCommandFormat(c,
```

Comment:
why not addReplyError()?

Reply:
i agree. let's keep `rejectCommand` used only in the pre-command checks (processCommand).
the main difference it has from `addReplyError` is that it's able to properly abort EXEC (in case the rejected command is EXEC)

Reply:
Applied.